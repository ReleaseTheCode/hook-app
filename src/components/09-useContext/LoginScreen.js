import React,{ useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const { setUser } = useContext(UserContext);
    const userObj = {
        id: 1,
        name: 'richter'
    } 

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button
                className='btn btn-primary'
                onClick={ ()=> setUser(userObj) }
            >
                Crear User
            </button>
        </div>
    )
}
