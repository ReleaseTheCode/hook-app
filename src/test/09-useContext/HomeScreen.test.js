import React from 'react';
import { mount, shallow } from 'enzyme';
import { HomeScreen } from '../../components/09-useContext/HomeScreen';
import { UserContext } from '../../components/09-useContext/UserContext';

describe('Test on <Homescreen/>', () => {
    const user = {
        name: 'Mr X',
        email: 'Mrx@test.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={ { user }}>
            <HomeScreen/> 
        </UserContext.Provider>
    );

    test('should render', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
});
