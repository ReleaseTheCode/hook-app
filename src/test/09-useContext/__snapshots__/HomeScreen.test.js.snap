// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Homescreen/> should render 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Mr X",
   "email": "Mrx@test.com"
}
  </pre>
</div>
`;
